packages/run-protocol/test/vaultFactory/test-liquidator.js

```diff
@@ -342,19 +342,73 @@ const makeDriver = async (t, initialPrice, priceBase) => {
     E(lender).getCollateralManager(aethBrand),
   ).getNotifier();
   let managerNotification = await E(managerNotifier).getUpdateSince();
-  /** @type {UserSeat<VaultKit>} */
-  let vaultSeat;
-  /** @type {VaultKit} */
-  let vault;
+
   /** @type {UserSeat<VaultKit>} */
   let lastSeat;
-  let notifier;
   let notification = {};
   let lastOfferResult;
+  const makeVaultDriver = async (collateral, debt) => {
+    /** @type {UserSeat<VaultKit>} */
+    const vaultSeat = await E(zoe).offer(
+      await E(lender).makeVaultInvitation(),
+      harden({
+        give: { Collateral: collateral },
+        want: { RUN: debt },
+      }),
+      harden({
+        Collateral: aethMint.mintPayment(collateral),
+      }),
+    );
+    const {
+      vault,
+      publicNotifiers: { vault: notifier },
+    } = await E(vaultSeat).getOfferResult();
+    t.truthy(await E(vaultSeat).hasExited());
+    return {
+      vault: () => vault,
+      vaultSeat: () => vaultSeat,
+      notification: () => notification,
+      close: async () => {
+        lastSeat = await E(zoe).offer(E(vault).makeCloseInvitation());
+        lastOfferResult = await E(lastSeat).getOfferResult();
+        t.is(
+          lastOfferResult,
+          'your loan is closed, thank you for your business',
+        );
+        t.truthy(await E(vaultSeat).hasExited());
+      },
+      checkNotify: async (phase, expected) => {
+        notification = await E(notifier).getUpdateSince();
+        trace(t, 'notify', notification);
+        t.is(notification.value.vaultState, phase);
+        expected && t.like(notification.value, expected);
+      },
+      awaitNotify: async (phase, expected) => {
```

Comment:
I like the check/await pair here but it's not obvious from names how they're different. They're both awaited internally so the `await` is a bit misnomer. The different is about waiting for the "next" in a sequence of updates so I'd suggest "next". Also would help tp have the `t.like` conveyed as what comes into `expected`.

So consider, `notifiedLike` and `notifiedNextLike`.

Reply:
In production code, I generally avoid having a boolean to give alternative behavior. But in this case, I think it would be simpler to have them in a single operation with a simple flag (or eventually optional sequence number). e.g.,
```
      vaultNotified: async (phase, expected, since) => {
        notification = await E(notifier).getUpdateSince(since === true ? notification.updateCount : since);
        t.is(notification.value.vaultState, phase);
        if (expected) {
          t.like(notification.value, expected);
        }
      },
```

Reply:
makes sense for test code. I don't like that there will be two independently optional arguments but again, test code. I would be explicit about that though with a jsdoc since the driver is an API (albeit an internal one).

some ideas for more clarity,
```
vaultNotified: async (phase, expectedLike, awaitNewNotification = false) => {
  const since = awaitNewNotification ? notification.updateCount : undefined;
```

also, falsy values can be expected so,
```
  // undefined can be expected but it's rare enough that we use it here to signal an unspecified parameter
  if (expectedLike === undefined)
```